### PR TITLE
[CALCITE-5655] Intermediate table alias should be different to avoid wrong field reference lookup in subquery remove phase

### PR DESCRIPTION
**Pull request by herunkang2018**

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java
Comment (herunkang2018) [inline core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java:6289]: These two test cases for expand project in is not affected by this issue, I add these to improve test converage.

**Comment by rubenada** [comment]
The fix looks good. There are some style violations to be corrected.

File: core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java
Comment (libenchao) [reply core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java:6289]: Thanks for the clarification, I think we can mention this in the commit message when we merging.

File: core/src/main/java/org/apache/calcite/rel/rules/SubQueryRemoveRule.java
Comment (libenchao) [inline core/src/main/java/org/apache/calcite/rel/rules/SubQueryRemoveRule.java:261]: Currently `RelBuilder#field` would return the first one when there is multiple results, I think we'd better to throw in this case. It would be a breaking change, what do you think about this @rubenada ?

**Comment by herunkang2018** [comment]
@rubenada Thanks for reminding it, I will fix it soon.

**Comment by herunkang2018** [comment]
@libenchao @rubenada All fixed, please help to review if you have time, thanks very much.

File: core/src/main/java/org/apache/calcite/rel/rules/SubQueryRemoveRule.java
Comment (rubenada) [reply core/src/main/java/org/apache/calcite/rel/rules/SubQueryRemoveRule.java:261]: Uhm... I think that might be a sensitive, breaking change. I'm not sure about it.
In any case, I think that should be discussed in a separated thread/ticket, and is out of the scope of the current PR.

**Comment by rubenada** [comment]
LGTM. I'd suggest to align the Jira title to the PRs title.

**Review by libenchao — APPROVED** [review]
+1

**Comment by herunkang2018** [comment]
@rubenada I have aligned the [Jira](https://issues.apache.org/jira/browse/CALCITE-5655) title to the PRs title. Please help to review if you have time.

**Comment by rubenada** [comment]
Thanks @herunkang2018 . PR looks in a good shape.
@libenchao since you assigned the PR to yourself, I'll let you finalize it and merge it, ok? (needs squashing commits into a single one).

**Review by rubenada — APPROVED** [review]

**Comment by libenchao** [comment]
@rubenada Sure, will merge this today. (Besides, since we have a mechanism that we give merits to the committer who merges the PR in our board report,  please feel free to do the merging if the PR is in good shape and everyone reaches consensus, the merit should not be scoped to the first one who assigns it to him/herself)

**Pull request closed without merging**
